test/spec/ApiConfigurationSpec.ts

```diff
@@ -0,0 +1,280 @@
+          <event name="XComponent.Common.Event.ApiProxy.InstanceUpdatedSubscription" id="3" />
+          <event name="XComponent.Common.Event.ApiProxy.InstanceUpdatedUnsubscription" id="4" />
+          <event name="XComponent.Common.Event.ApiProxy.SnapshotOptions" id="5" />
+          <event name="XComponent.Common.Event.DefaultEvent" id="6" />
+          <event name="XComponent.Common.Event.ExceptionEvent" id="7" />
+          <event name="XComponent.HelloWorld.UserObject.HelloWorldResponse" id="8" />
+          <event name="XComponent.HelloWorld.UserObject.SayHello" id="9" />
+        </events>
+        <stateMachines>
+          <stateMachine name="HelloWorldManager2" id="-829536630">
+            <states>
+              <State name="EntryPoint" id="0" />
+            </states>
+          </stateMachine>
+          <stateMachine name="HelloWorldResponse2" id="-343862281">
+            <states>
+              <State name="Start" id="0" />
+              <State name="Loop" id="1" />
+              <State name="Done" id="2" />
+            </states>
+          </stateMachine>
+        </stateMachines>
+      </component>
+      <component name="HelloWorld" id="-69981087">
+        <events>
+          <event name="XComponent.Common.Event.ApiProxy.ApiInitError" id="0" />
+          <event name="XComponent.Common.Event.ApiProxy.ApiInitSuccessful" id="1" />
+          <event name="XComponent.Common.Event.ApiProxy.CancelApiInit" id="2" />
+          <event name="XComponent.Common.Event.ApiProxy.InstanceUpdatedSubscription" id="3" />
+          <event name="XComponent.Common.Event.ApiProxy.InstanceUpdatedUnsubscription" id="4" />
+          <event name="XComponent.Common.Event.ApiProxy.SnapshotOptions" id="5" />
+          <event name="XComponent.Common.Event.DefaultEvent" id="6" />
+          <event name="XComponent.Common.Event.ExceptionEvent" id="7" />
+          <event name="XComponent.HelloWorld.UserObject.HelloWorldResponse" id="8" />
+          <event name="XComponent.HelloWorld.UserObject.SayHello" id="9" />
+        </events>
+        <stateMachines>
+          <stateMachine name="HelloWorldManager" id="-829536631">
+            <states>
+              <State name="EntryPoint" id="0" />
+            </states>
+          </stateMachine>
+          <stateMachine name="HelloWorldResponse" id="-343862282">
+            <states>
+              <State name="Start" id="0" />
+              <State name="Loop" id="1" />
+              <State name="Done" id="2" />
+            </states>
+          </stateMachine>
+        </stateMachines>
+      </component>
+    </components>
+  </codesConverter>
+</deployment>`;
+
+const parse = (xml: string) => {
+  const parser = new DefaultApiConfigurationParser();
+  return parser.parse(xml);
+};
+
+
+
+test("Test basic config parsing", () => {
+  return parse(xml)
+    .then(config => expect(config).toBeDefined());
+});
+
+test("Test parsing non xml format", () => {
```

Comment:
style remark: I would prefer to state what we expect from the test, instead of what it is doing.

i.e. Throws exception on non XML input.

This way we can spot easily on a failed tests report what the component should be doing that it is not doing anymore